### PR TITLE
NEW Add listing of bom realizable

### PR DESCRIPTION
**Pull request by inoveaconseil**

### CODE REVIEW

File: htdocs/mrp/index.php [hunk 1]
```diff
@@ -147,6 +148,71 @@
...
 print '<br>';
 
+$sql = "SELECT bom.rowid, bom.ref, bom.label ";
+$sql .= " FROM ".MAIN_DB_PREFIX."bom_bom AS bom, ".MAIN_DB_PREFIX."bom_bomline AS blin, ";
+$sql .= MAIN_DB_PREFIX."entrepot AS entpt, ".MAIN_DB_PREFIX."product_stock AS stk";
+$sql .= " WHERE bom.rowid = blin.fk_bom";
+$sql .= " AND blin.fk_product = stk.fk_product";
+$sql .= " AND stk.fk_entrepot = entpt.rowid";
+$sql .= " AND entpt.statut = 1";
+$sql .= " AND bom.entity = ".getEntity('bom');
```
Comment (eldy) [inline htdocs/mrp/index.php:158]: Must use " AND bom.entity IN (".getEntity('bom').")";

File: htdocs/mrp/index.php [hunk 2]
```diff
@@ -147,6 +148,71 @@
...
+$sql .= " AND bom.status = 1";
+$sql .= " AND bom.rowid NOT IN (";
+$sql .= " SELECT b.rowid";
+$sql .= " FROM ".MAIN_DB_PREFIX."bom_bom AS b";
+$sql .= " WHERE b.entity = ".getEntity('bom');
+$sql .= " AND b.rowid IN (";
+$sql .= " SELECT fk_bom";
+$sql .= " FROM ".MAIN_DB_PREFIX."bom_bomline AS bl";
+$sql .= " LEFT JOIN `llx_product_stock` AS st";
```
Comment (eldy) [inline htdocs/mrp/index.php:167]: Char ` is not portable and forbidden in SQL requests

File: htdocs/mrp/index.php [hunk 3]
```diff
@@ -147,6 +148,71 @@
...
+$sql .= " ON";
+$sql .= " bl.fk_product = st.fk_product";
+$sql .= " WHERE st.fk_product IS NULL OR bl.qty > (";
+$sql .= " SELECT SUM(s.reel)";
+$sql .= " FROM ".MAIN_DB_PREFIX."product_stock AS s,";
+$sql .= " ".MAIN_DB_PREFIX."entrepot AS ent";
+$sql .= " WHERE s.fk_product = bl.fk_product";
+$sql .= " AND s.fk_entrepot = ent.rowid";
+$sql .= " AND ent.statut = 1";
+$sql .= " AND ent.entity =".getEntity('bom');
```
Comment (eldy) [inline htdocs/mrp/index.php:177]: Use 
$sql .= " AND ent.entity IN (".getEntity('warehouse').")";

File: htdocs/mrp/index.php [hunk 4]
```diff
@@ -147,6 +148,71 @@
...
+$sql .= " AND blin.fk_product = stk.fk_product";
+$sql .= " AND stk.fk_entrepot = entpt.rowid";
+$sql .= " AND entpt.statut = 1";
+$sql .= " AND bom.entity = ".getEntity('bom');
+$sql .= " AND bom.status = 1";
+$sql .= " AND bom.rowid NOT IN (";
+$sql .= " SELECT b.rowid";
+$sql .= " FROM ".MAIN_DB_PREFIX."bom_bom AS b";
+$sql .= " WHERE b.entity = ".getEntity('bom');
+$sql .= " AND b.rowid IN (";
```
Comment (eldy) [inline htdocs/mrp/index.php:164]: You can avoid the 
b.rowid NOT IN (SELECT ... b.rowid IN (SELECT ...
with
b.rowid NOT IN (SELECT bom_bomline AS bl, bom as b ...  and a joint between bl and b.

File: htdocs/mrp/index.php [hunk 5]
```diff
@@ -147,6 +148,71 @@
 
 print '<br>';
 
+$sql = "SELECT bom.rowid, bom.ref, bom.label ";
+$sql .= " FROM ".MAIN_DB_PREFIX."bom_bom AS bom, ".MAIN_DB_PREFIX."bom_bomline AS blin, ";
```
Comment (eldy) [inline htdocs/mrp/index.php:152]: The table blin, entpt and stk here seems useless. Try to remove this table and their join completely.

File: htdocs/mrp/index.php (hunk 3 above)
Comment (hregis) [reply htdocs/mrp/index.php:177]: @eldy @inoveaconseil 

$sql .= " AND ent.entity IN (".getEntity('stock').")";

**Comment by eldy** [comment]
It seems this PR was set with "Discussion" or "PR to fix" tag, but no answer nor correction was provided since a long time. So request is automatically closed.
Please reopened and add a comment if you think this is an error.

**Pull request closed without merging**